### DISCUSSION TITLE
Setup local Supabase with cli after creating remote db with tables and RLS policies

### DISCUSSION THREAD

**Discussion by mikelyndon** [supabase/supabase#6287] [Questions] [answered]
Hi folks. I seem to be missing something in my understanding of the cli and how to link my local supabase setup with my remote account. 

I already have a bunch of tables with data and RLS policies setup in the remote host. What I want to do is continue development using a local image of that stuff. 
I have tried:
```bash
1. mkdir local-supabase
2. cd local-supabase
3. git init
4. supabase init
5. supabase link --project-ref <project-ref> 
```
I'm assuming `<project-ref>` is the unique string from my host `db.<project-ref>.supabase.co`
At this point the only change I can see is the `<project-ref>` has been added to a file in .temp/project-ref
Reading [some of the other discussions](https://github.com/supabase/supabase/discussions/3581#discussioncomment-1499323) and the docs has me thinking that the link command should create the initial migration file so that the local setup matches the remote host. No?

I've tried running
```bash
6. supabase db remote set 'postgresql://postgres:<password>@db.<project-ref>.supabase.co:5432/postgres'
```
But all I get is:
```bash
Error: The remote database was set up with a different Supabase CLI project. If you meant to reset the migration history to use a new Supabase CLI project:
1. Run DROP SCHEMA supabase_migrations CASCADE on the remote database,
2. Run supabase db remote set again,
3. Run supabase db remote commit.
```
I don't want to clear my remote db. I want to migrate the remote db schema to my local setup. That message seems to suggest that it will blow away my entire remote db setup.

So, how do I create a migration file that represents my remote db? I think that's what I need to get this working.

Thanks.

**Comment by mikelyndon** [comment, marked as answer]
Okay, problem existed between chair and computer. I incorrectly read that message about dropping the supabase_migrations schema, and I was paranoid because I nuked my db the last time I tried pushing a migration.

So the above steps do work.
```bash
1. mkdir local-supabase
2. cd local-supabase
3. git init
4. supabase init
5. supabase link --project-ref <project-ref> 
6. supabase db remote set 'postgresql://postgres:<password>@db.<project-ref>.supabase.co:5432/postgres'
7. supabase db remote commit
```

**Comment by KrownWealth** [comment]
is there a means of creating a new table from the supabase cli rather going to the remote project
